### DISCUSSION TITLE
UI freezes occasionally ... looking for hints.

### DISCUSSION THREAD

**Discussion by the-drunk-coder** [emilk/egui#1910] [Q&A]
Hi,

I have a heavily customized text editor widget that's based on egui's text editor widget:

https://github.com/the-drunk-coder/megra.rs/blob/main/src/editor/livecode_text_edit.rs

It seems like sometimes the UI freezes when not being used for a while (2-3 min), even though the rest of the program keeps running, that is, the real-time thread that generates the sound and the threads that do the scheduling remain active. 

There's no visible error on the console.

Anyone's got a hint how to trace this problem ?

**Comment by emilk** [comment]
Attach a debugger and check the stack trace?

**Comment by zicklag** [comment]
The most common cause of freezes for me while using egui is a deadlock on the context. But it's not the kind of thing I would expect to run into after minutes of inactivity...
